### PR TITLE
fix: proper links handling while fetching orgs

### PR DESCRIPTION
**Pull request by bartoszmajsak**

### CODE REVIEW

File: _ext/repository.rb [hunk 1]
```diff
@@ -278,6 +276,22 @@ def execute(site)
...
+        def parse_link_headers(response)
+          links = Hash.new
+          if response.headers[:link].nil?
+            return links
+          end
+          parts = response.headers[:link].split(', ')
+
+          parts.each do |part, index|
+            section = part.split(';')
+            url = section[0][/<(.*)>/,1]
```
Comment (dipak-pawar) [inline _ext/repository.rb]: we should check length for section here. otherwise we'll get `section[0]` or `section[1]` as `nil`

Comment (bartoszmajsak) [reply _ext/repository.rb]: I think in such a case we would have invalid `link` header according to the [spec](https://tools.ietf.org/html/rfc5988) or am I missing something? We can always be defensive though.

File: _ext/repository.rb [hunk 2]
```diff
@@ -185,11 +185,9 @@ def execute(site)
...
                 end
               }
-              if org_repos_data.headers[:link] =~ /<(.*)>.+rel="next",/
-                org_url = $1
-              else
-                more_pages = false
-              end
+              links = parse_link_headers(org_repos_data)
+              org_url = links[:next]
+              more_pages = (links.has_key?(:next) and links.has_key?(:last) and (links[:next] != links[:last]))
```
Comment (MatousJobanek) [inline _ext/repository.rb]: why does the next link cannot equal to the last link? I guess that this should be the last page (link) we should check. 
I believe that checking for the presence of the next link should be enough:
`more_pages = links.has_key?(:next)`

File: _ext/repository.rb (hunk 1 above)
Comment (dipak-pawar) [reply _ext/repository.rb]: specification of link header helps. Thanks

File: _ext/repository.rb (hunk 2 above)
Comment (bartoszmajsak) [reply _ext/repository.rb]: There is always `next` and that was the problem of infinite loop.

Comment (bartoszmajsak) [reply _ext/repository.rb]: I was wrong (let's blame lack of coffee ;)). Thanks, it's fixed.

**Pull request merged**